internal/conf/jwk.go

```diff
@@ -0,0 +1,192 @@
+package conf
+
+import (
+	"crypto/ecdh"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+type JwtKeysDecoder map[string]JwkInfo
+
+type JwkInfo struct {
+	PublicKey  jwk.Key `json:"public_key"`
+	PrivateKey jwk.Key `json:"private_key"`
+}
+
+// Decode implements the Decoder interface
+// which transforms the keys stored as der binary strings into jwks
+func (j *JwtKeysDecoder) Decode(value string) error {
+	data := make([]map[string]interface{}, 0)
+	if err := json.Unmarshal([]byte(value), &data); err != nil {
+		return err
+	}
+
+	config := JwtKeysDecoder{}
+	for _, key := range data {
+		bytes, err := json.Marshal(key)
+		if err != nil {
+			return err
+		}
+		privJwk, err := jwk.ParseKey(bytes)
+		if err != nil {
+			return err
+		}
+		pubJwk, err := jwk.PublicKeyOf(privJwk)
+		if err != nil {
+			return err
+		}
+
+		// all public keys will be used for signature verification
+		if pubJwk.KeyUsage() == "enc" {
+			if err := pubJwk.Set(jwk.KeyUsageKey, "sig"); err != nil {
+				return err
+			}
+		}
+		config[pubJwk.KeyID()] = JwkInfo{
+			PublicKey:  pubJwk,
+			PrivateKey: privJwk,
+		}
+	}
+	*j = config
+	return nil
+}
+
+func (j *JwtKeysDecoder) Validate() error {
+	// Validate performs _minimal_ checks if the data stored in the key are valid.
+	// By minimal, we mean that it does not check if the key is valid for use in
+	// cryptographic operations. For example, it does not check if an RSA key's
+	// `e` field is a valid exponent, or if the `n` field is a valid modulus.
+	// Instead, it checks for things such as the _presence_ of some required fields,
+	// or if certain keys' values are of particular length.
+	//
+	// Note that depending on th underlying key type, use of this method requires
+	// that multiple fields in the key are properly populated. For example, an EC
+	// key's "x", "y" fields cannot be validated unless the "crv" field is populated first.
+	for _, key := range *j {
+		if err := key.PrivateKey.Validate(); err != nil {
+			return err
+		}
+		// symmetric keys don't have public keys
+		if key.PublicKey != nil {
+			if err := key.PublicKey.Validate(); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func GetSigningJwk(config *JWTConfiguration) (*jwk.Key, error) {
+	if config.Keys == nil {
+		symmetricJwk, err := getSymmetricKey(config)
+		if err != nil {
+			return nil, err
+		}
+		return symmetricJwk, nil
+	}
+	for _, key := range config.Keys {
+		if key.PrivateKey.KeyUsage() == "enc" {
+			return &key.PrivateKey, nil
+		}
```

Comment:
This seems off... why are you looking for an `enc` (encryption) key?

Reply:
we use the `enc` key to denote the private key used to sign JWTs

Reply:
Hm that's not right. [`use`](https://datatracker.ietf.org/doc/html/rfc7517#page-6) signals use for encryption (which is not signing!) or signing.

For example, elliptic curve cryptosystems do not have encryption support, while [RSA does](https://en.wikipedia.org/wiki/RSA_(cryptosystem)#Encryption). Because of the way RSA signatures / encryption work, you must not use the same RSA key for both signatures and encryption which can lead to private key compromise. Therefore the `use` attribute signals what the key is supposed to be used for.

Let's not misuse the standard. The `use` for JWTs is always `sig` and never `enc`. You need to disambiguate in code whether the key is _private_ which means that signatures can be created and verfied or _public_ which means that signatures can only be verified.

Reply:
decided to use `key_ops` to represent the JWK used for signing instead